sparse/lib/udev/rules.d/998-droid-system.rules

```diff
@@ -54,10 +54,11 @@ SUBSYSTEM=="platform", KERNEL!="", DEVPATH!="/devices/platform/*", DEVPATH!="/de
 
 LABEL="bootdevice_end"
 
-# Create the partition symlinks.
-ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_NAME}=="?*", SYMLINK+="block/platform/$env{PLATFORM_FOLDER}/$env{PLATFORM_DEVICE}/by-name/$env{ID_PART_ENTRY_NAME}"
-ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_NAME}=="?*", SYMLINK+="block/platform/$env{PLATFORM_FOLDER}/$env{PLATFORM_DEVICE}/by-num/p$env{ID_PART_ENTRY_NUMBER}"
-ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_NAME}=="?*", SYMLINK+="block/platform/$env{PLATFORM_FOLDER}/$env{PLATFORM_DEVICE}/$name"
+# Create the partition symlinks. Use external command for linking to prevent breakage with bootdevice symlink
+ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_NAME}=="?*", RUN+="/bin/mkdir -p /dev/block/platform/$env{PLATFORM_FOLDER}/$env{PLATFORM_DEVICE}/by-name"
+ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_NAME}=="?*", RUN+="/bin/ln -s /dev/block/$name /dev/block/platform/$env{PLATFORM_FOLDER}/$env{PLATFORM_DEVICE}/by-name/$env{ID_PART_ENTRY_NAME}"
+ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_NAME}=="?*", RUN+="/bin/ln -s /dev/block/$name /dev/block/platform/$env{PLATFORM_FOLDER}/$env{PLATFORM_DEVICE}/by-num/p$env{ID_PART_ENTRY_NUMBER}"
+ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_NAME}=="?*", RUN+="/bin/ln -s /dev/block/$name /dev/block/platform/$env{PLATFORM_FOLDER}/$env{PLATFORM_DEVICE}/$name"
 
```

Comment:
Why that that change?  Udev seems to use relative symlinks for /dev/disk/by-*/* symlinks.

Reply:
That is because we have that bootdevice folder symlink which breaks the links under it in some cases.

Reply:
OK can you replace the RUN part with SYMLINK?

Reply:
SYMLINK will create relative symlinks that's why the external command is used to create absolute symlinks.

bootctl will not work with relative symlinks.